packages/eslint-plugin/src/rules/no-misused-promises.ts

```diff
@@ -674,10 +672,5 @@ function isVoidReturningFunctionType(
  */
 function returnsThenable(checker: ts.TypeChecker, node: ts.Node): boolean {
   const type = checker.getApparentType(checker.getTypeAtLocation(node));
-
-  if (anySignatureIsThenableType(checker, node, type)) {
-    return true;
-  }
-
-  return false;
+  return anySignatureIsThenableType(checker, node, type);
```

Comment:
https://github.com/typescript-eslint/typescript-eslint/pull/8534 is going to add support for unions. And it looks like it's going to make changes exactly on these very lines. So, to avoid merge conflicts, what do you think about reverting these changes in favor of https://github.com/typescript-eslint/typescript-eslint/pull/8534

Reply:
makes sense! reverting and resolving conflicts